src/python/genny/genny_auto_tasks.py

```diff
@@ -0,0 +1,74 @@
+import os
+import re
+import sys
+import subprocess
+import textwrap
+
+from shrub.config import Configuration
+from shrub.command import CommandDefinition
+from shrub.variant import TaskSpec
+
+def to_snake_case(str):
+	"""
+	Converts str to snake_case, useful for generating test id's
+	From: https://stackoverflow.com/questions/1175208/elegant-python-function-to-convert-camelcase-to-snake-case
+	:return: snake_case version of str.
+	"""
+	s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', str)
+	return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
+
+def modified_workload_files():
+	"""
+	Returns a list of filenames for workloads that have been modified according to git, relative to origin/master.
+	:return: a list of filenames in the format subdirectory/Task.yml
+	"""
+	out = subprocess.check_output('git diff --name-only `git merge-base HEAD origin/master` -- ../workloads/', shell=True)
```

Comment:
let's also have a test of this function that mocks out `subprocess.check_output`. In particular I'm interested in the behavior when 

1. there are multiple workload files modified
2. there are non-workload files modified
3. somehow some or part of the command fails
4. there are no files modified
5. a workload file is `git rm`ed (deleted)
6. a workload file is `git mv`ed (renamed)
7. attributes (e.g. `chmod +x` are applied to workload files

You can quickly figure out what the output of the command is for mock-purposes by just modifying some files and copy/pasting the output into the test-case.

I would also have similar test-cases for your other function and assert the json generated (or lack of it) rather than just the happy-case.

Use your best-judgment about what test-cases in my list above are actually distinct and worthy of their own cases. I don't really know what the expected-beavhior should be for some of these so I'll let you use your best-judgment but these are all at least worth considering.

Reply:
Thanks, these are good cases and I didn't consider all of them. A couple questions:

- If a workload file was renamed should that be included in the patch workloads to run?
- If a workload file is added, but the new file is not committed or staged for commit, should that workload be included?
- Does the code have to be resilient against a case where there is a non-workload .yml file in the workloads directory? (I lean towards no because it seems excessive to me for this script to have to read and verify whether each file is a workload file)

Reply:
> If a workload file was renamed should that be included in the patch workloads to run?
Yep. It should be covered by your existing git filter.

> If a workload file is added, but the new file is not committed or staged for commit, should that workload be included?

Evergreen will apply your changes as a git patch but will not commit or stage the patch, so you ONLY need to consider checking for unstaged changes.

> Does the code have to be resilient against a case where there is a non-workload .yml file in the workloads directory? (I lean towards no because it seems excessive to me for this script to have to read and verify whether each file is a workload file)

Definitely not. If someone commits a non-workload or an ill formatted workload, patch builds are not expected to work, even for manually created tasks.

Reply:
I added the R option to the --diff-filter so that renamed files are included!